XenAdmin/Diagnostics/Problems/PoolProblem/PoolHasVmsWithDmcProblem.cs

```diff
@@ -0,0 +1,146 @@
+ * that the following conditions are met: 
+ * 
+ * *   Redistributions of source code must retain the above 
+ *     copyright notice, this list of conditions and the 
+ *     following disclaimer. 
+ * *   Redistributions in binary form must reproduce the above 
+ *     copyright notice, this list of conditions and the 
+ *     following disclaimer in the documentation and/or other 
+ *     materials provided with the distribution. 
+ * 
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND 
+ * CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, 
+ * INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF 
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE 
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR 
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, 
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, 
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR 
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS 
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, 
+ * WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING 
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE 
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF 
+ * SUCH DAMAGE.
+ */
+
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using System.Windows.Forms;
+using XenAdmin.Actions;
+using XenAdmin.Commands;
+using XenAdmin.Core;
+using XenAdmin.Diagnostics.Checks;
+using XenAdmin.Dialogs;
+using XenAPI;
+
+namespace XenAdmin.Diagnostics.Problems.PoolProblem
+{
+    class PoolHasVmsWithDmcProblem : ProblemWithMoreInfo
+    {
+        private readonly Pool _pool;
+        private readonly List<VM> _vms;
+        protected readonly Control _control;
+
+        public PoolHasVmsWithDmcProblem(Control control, Check check, Pool pool, List<VM> vms)
+            : base(check)
+        {
+            _control = control;
+            _pool = pool;
+            _vms = vms;
+        }
+
+        public override string Title => Description;
+
+        public override string Description =>
+            string.Format(Messages.DMC_REMOVAL_SHORT, _pool.Name(),
+                string.Format(Messages.STRING_SPACE_STRING, BrandManager.ProductBrand, BrandManager.ProductVersionPost82));
+
+        public override string Message =>
+            string.Format(Messages.DMC_REMOVAL_LONG_ERROR,
+                string.Format(Messages.STRING_SPACE_STRING, BrandManager.ProductBrand, BrandManager.ProductVersionPost82),
+                BrandManager.BrandConsole);
+
+        protected override AsyncAction CreateAction(out bool cancelled)
+        {
+            Program.Invoke(_control, delegate
+            {
+                using (var dlg = new ErrorDialog(Message,
+                           new ThreeButtonDialog.TBDButton(Messages.DMC_DISABLE, DialogResult.OK, ThreeButtonDialog.ButtonType.ACCEPT, true),
+                           ThreeButtonDialog.ButtonCancel)

```

Comment:
I'd set `Cancel` as the default button since the action shuts down VMs.

![image](https://user-images.githubusercontent.com/32554698/156539445-677352ef-c635-4c20-a370-0e44e8cfd215.png)
```suggestion
                using (var dlg = new ErrorDialog(Message,
                           new ThreeButtonDialog.TBDButton(Messages.DMC_DISABLE, DialogResult.OK, ThreeButtonDialog.ButtonType.ACCEPT),
                           new ThreeButtonDialog.TBDButton(Messages.CANCEL, DialogResult.Cancel, ThreeButtonDialog.ButtonType.CANCEL, true))
```

or at least remove pre-selection

```suggestion
                using (var dlg = new ErrorDialog(Message,
                           new ThreeButtonDialog.TBDButton(Messages.DMC_DISABLE, DialogResult.OK, ThreeButtonDialog.ButtonType.ACCEPT),
                           ThreeButtonDialog.ButtonCancel)
```